job-server/test/spark.jobserver/auth/SJSAuthenticatorSpec.scala

```diff
@@ -0,0 +1,138 @@
+package spark.jobserver.auth
+
+import org.scalatest.{ FunSpecLike, FunSpec, BeforeAndAfter, BeforeAndAfterAll, Matchers }
+import org.apache.shiro.config.IniSecurityManagerFactory
+import org.apache.shiro.mgt.DefaultSecurityManager
+import org.apache.shiro.mgt.SecurityManager
+import org.apache.shiro.realm.Realm
+import org.apache.shiro.SecurityUtils
+import org.apache.shiro.config.Ini
+import spark.jobserver.auth._
+import spray.routing.authentication.UserPass
+
+import spray.routing.directives.AuthMagnet
+import spray.routing.{ HttpService, Route, RequestContext }
+import spray.http.StatusCodes
+import spray.testkit.ScalatestRouteTest
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ Await, ExecutionContext, Future }
+import org.slf4j.LoggerFactory
+
+class SJSAuthenticatorSpec extends HttpService with SJSAuthenticator with FunSpecLike
+  with ScalatestRouteTest with Matchers with BeforeAndAfter with BeforeAndAfterAll {
+
+  def actorRefFactory = system
+
+  //edit this with your real LDAP server information, just remember not to 
+  // check it in....
```

Comment:
No, i just left this in in case someone wants to test this against a real LDAP server
The actual test uses the simple authorization procedure provided by the Shiro example.